update settings.xml to the new format. Add language and region selection for TMDB

Coffelius opened the pull request:
This pull request introduces the same language and region selection for TMDb as used in the TMDb Helper add-on. This enhancement addresses issues with selecting languages, which can result in different titles depending on the country.

To implement this, I updated the format of settings.xml. The reasons for this update are as follows:
	•	Compatibility: The current format was deprecated in Kodi 19.
	•	Improved Usability: By utilizing the help attribute, inline documentation is provided, enabling users to understand each setting without relying on external guides or trial and error.
	•	Enhanced Functionality: The new format supports multiple selections in lists, which opens the door for additional improvements, such as removing the need for custom dialogs when choosing Torrentio providers in the future. It also supports the list control required to select language and region.

Coffelius commented:
@Sam-Max do you think it would be better to merge this to develop branch?

Sam-Max commented:
> @Sam-Max do you think it would be better to merge this to develop branch?

Yes, sure

Sam-Max commented:
> @Sam-Max do you think it would be better to merge this to develop branch?

Not necessary, keep working on the feature branch. I will delete the develop branch

Sam-Max commented:
It seems some settings needs a default value like for example jackgram_host or jackett_host in order to show

Coffelius commented:
> It seems some settings need a default value like for example jackgram_host or jackett_host in order to show up

This issue was resolved by adding the following constraint for settings without a default value:
 `<constraints><allowempty>true</allowempty></constraints>`

Additionally, this PR introduces language and region selection for TMDB. This improvement addresses issues with titles appearing in the wrong language or selecting titles from random countries that share the same language.

Sam-Max commented:
> > It seems some settings need a default value like for example jackgram_host or jackett_host in order to show up
> 
> This issue was resolved by adding the following constraint for settings without a default value: `<constraints><allowempty>true</allowempty></constraints>`
> 
> Additionally, this PR introduces language and region selection for TMDB. This improvement addresses issues with titles appearing in the wrong language or selecting titles from random countries that share the same language.

On the TMDB language selection, there seems to be some strings missing for some languages

Coffelius commented:
> > > It seems some settings need a default value like for example jackgram_host or jackett_host in order to show up
> > 
> > 
> > This issue was resolved by adding the following constraint for settings without a default value: `<constraints><allowempty>true</allowempty></constraints>`
> > Additionally, this PR introduces language and region selection for TMDB. This improvement addresses issues with titles appearing in the wrong language or selecting titles from random countries that share the same language.
> 
> No need to set default value, good solution. On the TMDB language selection, there seems to be some strings missing for some languages

good catch. added the missing translations

Sam-Max commented:
> > > > It seems some settings need a default value like for example jackgram_host or jackett_host in order to show up
> > > 
> > > 
> > > This issue was resolved by adding the following constraint for settings without a default value: `<constraints><allowempty>true</allowempty></constraints>`
> > > Additionally, this PR introduces language and region selection for TMDB. This improvement addresses issues with titles appearing in the wrong language or selecting titles from random countries that share the same language.
> > 
> > 
> > No need to set default value, good solution. On the TMDB language selection, there seems to be some strings missing for some languages
> 
> good catch. added the missing translations

I don't see more bugs or things to fix, I think it is fine to merge this PR
